lib/irb/ext/tracer.rb

```diff
@@ -42,37 +32,7 @@ class Context
     #
     # See +lib/tracer.rb+ for more information.
     def use_tracer=(opt)
-      if opt
-        Tracer.set_get_line_procs(@irb_path) {
-          |line_no, *rests|
-          @io.line(line_no)
-        }
-      elsif !opt && @use_tracer
-        Tracer.off
-      end
       @use_tracer=opt
     end
   end
-
-  class WorkSpace
-    alias __evaluate__ evaluate
-    # Evaluate the context of this workspace and use the Tracer library to
-    # output the exact lines of code are being executed in chronological order.
-    #
-    # See +lib/tracer.rb+ for more information.
-    def evaluate(context, statements, file = nil, line = nil)
-      if context.use_tracer? && file != nil && line != nil
-        Tracer.on
-        begin
-          __evaluate__(statements, file, line)
-        ensure
-          Tracer.off
-        end
-      else
-        __evaluate__(statements, file || __FILE__, line || __LINE__)
-      end
-    end
```

Comment:
I could not figure out what this is supposed to do. The cardinality seems wrong: https://github.com/ruby/irb/blob/d8fb3246be699dea102a23410fe7555ae2905f9c/lib/irb/workspace.rb#L116

After correcting the cardinality we can see that this method is invoked, and always falls into the "else" case.

If there is a use case we can write a test for, then we can bring this back.

Reply:
If the tracer is on, we may be able to do something like:

```rb
CallTracer.new(pattern: /^#{Regexp.quote(@CONF[:IRB_LIB_PATH])}/).start do
  __evaluate__(statements, file, line)
end
```